### PR TITLE
Enhance token endpoint by setting cookie

### LINKED ISSUE #542: Enhance /tokens endpoint to optionally set cookie
**Is your feature request related to a problem? Please describe.**

Currently the UI keeps the access-token for the logged in user as part of the application state. There are at least a couple of concerns with this
- Malicious javascript could potentially retrieve the access-token
- The application state is lost on browser refresh, requiring the user to re-authenticate

**Describe the solution you would like**

Extend the `/tokens` endpoint to also set a Cookie with the generated access-token. And extend server authorization to look at the cookie in addition to the `Authorization` header.

The pieces are as follows:

1.  Extend `AuthService.grantToken()` to look for the request query parameter `ext=cookie`. If the query parameter and value exists, the `grantToken()` should set the cookie `UC_TOKEN` with the value of the access token it has generated. This should be an `HttpOnly` token so that Javascript does not have access to it.
2. Extend `AuthDecorator.serve()` so that if it does not find an `Authorization` header, it then checks for the `UC_TOKEN` cookie to get the access-token to validate and use as for the authorized principal for the request. 

**Additional context**

### PR DESCRIPTION
**Pull request by sudharshanraja-db**
**PR Checklist**

- [x] A description of the changes is added to the description of this PR.
- [x] If there is a related issue, make sure it is linked to this PR.
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added or modified a feature, documentation in `docs` is updated

**Description of changes**

- Set token as only http only cookie in the response header if query param ext=cookie is set
- Cookie default age is set to 5 days and this is a configurable parameter in server.properties
- Authorize user with token in cookie if bearer token is missing

Closes #542

### CODE REVIEW

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java
Comment (creechy) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:157]: @jamieknight-db might comment on any specific cookie settings she needs for the UI, looks like `secureBuilder()` sets `secure(true).httpOnly(true).sameSite("Strict")` which feels right.

Comment (creechy) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:157]: Also, I was thinking that by default, lets still be more Oauth2 compliant and not set a header, but when we see the query parameter `ext=cookie` then set it. So @jamieknight-db would call `auth/tokens?ext=cooke` to get a response that sets a cookie.

File: server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java
Comment (creechy) [inline server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java:207]: Hmm, I'm guessing this is here, because you started your PR branch off of something other than `main`? it should probably clear up once I merge in the SCIM PR you did.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java
Comment (jamieknight-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:157]: I might be misunderstanding, but I don't plan to utilize the cookie/token at all on the FE. I plan to call a `currentUser` endpoint and a `login` endpoint, both of which will hopefully send me back a `user` object upon success (if auth is enabled).

Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java:157]: Sorry yeah thanks for the reminder, yeah you'll let the browser take care of it. That does remind me, we should probably set a cookie timeout, say 5 days or something like that?

File: server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java]: Yes addressed it

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: @creechy @jamieknight-db  what is the recommended timeout value?

Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: Add the condition to set based on the query param

File: server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java:207]: yeah i see one pending PR ,  i can also rebase once you merge

**Comment by sudharshanraja-db** [comment]
Attaching the screenshots of cookie being set and different max age honoured based on config 
[cookie_browser.pdf](https://github.com/user-attachments/files/17345104/cookie_browser.pdf)
cookie_timeout_diff_values

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java
Comment (jamieknight-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: There isn't really a recommended timeout value, it just depends on the use case. I think @creechy's suggestion of 5 days or 7 days is good 👍  thank you!

Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthService.java]: @jamieknight-db  yeah we have kept it as default 5 days and this value is configurable(can be changed in server.properties)

File: server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java]: fixed

File: server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java
Comment (sudharshanraja-db) [reply server/src/main/java/io/unitycatalog/server/service/AuthDecorator.java]: fixed

File: server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java
Comment (creechy) [reply server/src/main/java/io/unitycatalog/server/service/Scim2UserService.java]: You should be able to rebase whenever. The PR is for merging into the v0.2 patch branch.

File: server/src/main/java/io/unitycatalog/server/service/AuthService.java
Comment (creechy) [inline server/src/main/java/io/unitycatalog/server/service/AuthService.java:174]: Lets just have the `server.properties` value be the full string the `parse()` method expects. Spring users for example, should be familiar with the full syntax.

File: etc/conf/server.properties
Comment (creechy) [inline etc/conf/server.properties:16]: Lets just require the full syntax string. See my other comment. You can reference `Duration.parse()` in the comment above for context.

**Review by creechy — APPROVED** [review]
Looks good.

**Comment by jamieknight-db** [comment]
@sudharshanraja-db is it ok if I merge this? Or would you like to wait for more reviews?

**Comment by sudharshanraja-db** [comment]
> @sudharshanraja-db is it ok if I merge this? Or would you like to wait for more reviews?

@creechy  i saw you have reviewed can u comment if further reviews are required

**Review by dennyglee — APPROVED** [review]

**Comment by creechy** [comment]
> @sudharshanraja-db is it ok if I merge this? Or would you like to wait for more reviews?

@jamieknight-db @sudharshanraja-db I think this can be merged now.

**Pull request merged**
